include/deal.II/grid/grid_in.h

```diff
@@ -567,6 +569,65 @@ class GridIn
               const double       tol                              = 1e-12,
               const bool         ignore_unsupported_element_types = true);
 
+  /**
+   * Read in a mesh stored in the ExodusII file format.
+   *
+   * ExodusII is a feature-rich file format that supports many more features
+   * (like node sets, finite element fields, quality assurance data, and more)
+   * than most other grid formats supported by this class. Many of these
+   * features do not have equivalent representations in deal.II and are
+   * therefore not supported (for example, deal.II does not assign degrees of
+   * freedom directly to nodes, so data stored in a nodal format is not loaded
+   * by this function). At the current time only the following information is
+   * extracted from the input file:
+   *
+   * <ol>
+   *   <li>Block ids: the block id of an element is loaded as its material
+   *     id.</li>
+   *   <li>Elements and vertices: the core geometric information stored in the
+   *     ExodusII file populates the attached Triangulation object. Higher-order
+   *     elements are automatically truncated to lower-order elements since
+   *     deal.II does not support this feature (e.g., there is no equivalent to
+   *     the <code>QUAD9</code> element in deal.II since all quadrilaterals have
+   *     four vertices and additional geometric information is either stored in
+   *     a Manifold or something like MappingQEulerian).</li>
```

Comment:
It would be pretty easy to write some more code that loads the element connectivity again and establishes a mapping between the vertex numbers and the dofs we assign to extract this information.

We might want to *save* the connectivity data at this point so that we can extract this easily which is a good argument for returning a `struct`.

Reply:
Just so we remember for later - this is basically the feature proposed in #10163 and it would be great if we could generalize the implementation to support multiple backends (like ExodusII).